spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DatastoreEntityConverter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core.convert;
+
+import com.google.cloud.datastore.Entity;
+
+import org.springframework.data.convert.EntityReader;
+import org.springframework.data.convert.EntityWriter;
+
+/**
+ * An interface for converting objects to Datastore Entities and vice versa.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+public interface DatastoreEntityConverter extends
+		EntityReader<Object, Entity>, EntityWriter<Object, Entity.Builder> {
```

Comment:
What's the benefit of implementing reader and writer in a single class?

Reply:
There is in fact a Spring Data `EntityConverter` class that combines both reader and writer, but we aren't implementing that because it requires the source of the read to be the same type as the type of the write. For example I believe the mongodb one implements that converter interface since it has Document as the source and sink type. 

Having them together in one class also requires that want bi-directional conversion, which I think is always going to be true here.

Do you have a reason why they should be separate?

Reply:
Let's keep them together for now, it looks like it logically belongs together. We can always change if needed.